README.md

```diff
@@ -114,50 +114,43 @@ Set up a `SignInWithAppleButton` via XML:
     app:sign_in_with_apple_button_cornerRadius="4dp" />
 ```
 
-In your Activity, create the `SignInWithAppleService`, implement `SignInWithAppleClient`, and configure the button with both:
+In your Activity, create the `SignInWithAppleService`, and configure the button with it:
 
 ```kotlin
-override fun onStart() {
-    super.onStart()
+override fun onCreate(savedInstanceState: Bundle?) {
+    ...
 
     val service = SignInWithAppleService(
         clientId = "com.your.client.id.here",
         redirectUri = "https://your-redirect-uri.com/callback",
         scope = "email"
     )
 
-    val client: SignInWithAppleClient = this
-
     val signInWithAppleButton = findViewById(R.id.sign_in_with_apple_button)
-    signInWithAppleButton.configure(service, client)
-}
-
-// SignInWithAppleClient
-
-override fun getFragmentManagerForSignInWithApple(): FragmentManager {
-    return supportFragmentManager
-}
-
-override fun onSignInWithAppleSuccess(authorizationCode: String) {
-    // Handle success
-}
-
-override fun onSignInWithAppleFailure(error: Throwable) {
-    // Handle failure
+    signInWithAppleButton.configure(supportFragmentManager, service, client) { result ->
+        when (result) {
+            is SignInWithAppleResult.Success -> {
+                // Handle success
+            }
+            is SignInWithAppleResult.Failure -> {
+                // Handle failure
+            }
+            is SignInWithAppleResult.Cancel -> {
+                // Handle user cancel
+            }
+        }
+    }
 }
-```
-
-> It's necessary to call `configure()` after device rotation. Configuring the button in `onStart()` worked well for us in all cases.
```

Comment:
Not a necessary explanation? Is this just common knowledge?

Reply:
I think it's clear enough with how the api currently is. The only way it won't be set up correctly is if you configured it outside of the activity lifecycle somehow but that would be weird.